permissions_validators/src/lib.rs

```diff
@@ -98,6 +98,65 @@ macro_rules! try_into_or_exit {
     };
 }
 
+macro_rules! declare_token {
+    (
+        $(#[$outer_meta:meta])* // Structure attributes
+        $ident:ident {          // Structure definiton
+            $(
+                $param_name:ident ($param_string:literal): $param_typ:ty
+             ),* $(,)? // allow trailing comma
+        },
+        $string:tt // Token name
+    ) => {
+
+        $(#[$outer_meta])*
+        ///
+        /// A wrapper around [PermissionToken](iroha_data_model::permissions::PermissionToken).
+        #[derive(iroha_schema::IntoSchema)]
+        pub struct $ident
+        where $($param_typ: Into<Value>,)* {
+            $(
+                $param_name : $param_typ
+             ),*
+        }
+
+        impl $ident {
+            /// Get associated [`PermissionToken`](iroha_data_model::permissions::PermissionToken) name.
+            pub fn name() -> &'static Name {
+                static  NAME: once_cell::sync::Lazy<Name> =
+                    once_cell::sync::Lazy::new(|| $string.parse().expect("Tested. Works."));
+                &NAME
+            }
+
+            $(
+              #[doc = concat!("Get `", stringify!($param_name), "` parameter name")]
+              pub fn $param_name() -> &'static Name {
```

Comment:
```suggestion
              pub fn $param_name() -> $param_typ {
```

if you ask me this should return the value stored in the token corresponding to the parameter name

Reply:
I made those methods to be able to replace _PARAM_NAME statics, that why it's `Name`
I think struct fields can be made public actually, since I don't impose any bounds on their values in the constructor anyway